### ISSUE TITLE
Doc:  Clarify host (saas) or url (smas) requirements

### ISSUE THREAD

**Issue by ppf2** [logstash-plugins/logstash-output-elastic_app_search#8]
https://www.elastic.co/guide/en/logstash/master/plugins-outputs-elastic_app_search.html

The documentation is not that clear on host and url requirements.  Under the [table](https://www.elastic.co/guide/en/logstash/master/plugins-outputs-elastic_app_search.html#plugins-outputs-elastic_app_search-options), it shows that both host and url are optional parameters.

But host is actually required for SaaS, and url is required for SMAS (self-managed).  

In the host description of the doc, it also only specifies it as being an option for use with SMAS.   It will be helpful to clarify the options that are specific to SaaS, SMAS or shared by both, and to tag the options that are required for each deployment type.

**Comment by jsvd** [comment]
> it shows that both host and url are optional parameters

yeah I agree this needs clarification, as in the strict sense these settings aren't required individually, but you have to set one of them.

>  to tag the options that are required for each deployment type.

agreed.
